lib/vdsm/virt/thinp.py

```diff
@@ -536,24 +606,26 @@ def _extend_volume_completed(self, volInfo):
                 with clock.run("refresh-destination-volume"):
                     self._vm.refresh_destination_volume(volInfo)
 
-            with clock.run("refresh-volume"):
-                self._vm.refresh_volume(volInfo)
+            # After a volume was refreshed, the monitor thread may wake up and
+            # trigger unwanted extend to the next chunk size. Lock the drive
+            # during refresh until we set a new threshold.
```

Comment:
The fact that I'm asking again myself what will happen with the refreshing when the lock times out although I've already asked about it, indicates that it deserves an additional comment here.

Reply:
Sure, will explain what happen if locking times out.